daemons/gptp/common/ieee1588port.cpp

```diff
@@ -598,7 +598,8 @@ void IEEE1588Port::processEvent(Event e)
 
 	case STATE_CHANGE_EVENT:
 		if (!automotive_profile) {       // BMCA is not active with Automotive Profile
-			if ( clock->getPriority1() != 255 ) {
+			if ( clock->getPriority1() != 255 && qualified_announce != NULL ) {
```

Comment:
Agree with @pinealservo there's not really any way I could see that qualified_announce would be NULL. If you want to check for NULL, I think it would be better to check the value of EBest, but still not necessary.